extensions/wikia/WikiaInYourLang/modules/ext.wikiaInYourLang.js

```diff
@@ -155,6 +160,43 @@ require(
 			tracker.track(trackingParams);
 		}
 
+		function getWIYLRequestSentKey() {
+			return 'wikiaInYourLangRequestSent' + retrieveLinkTitle() + cacheVersion;
+		}
+
+		function getWIYLNotificationShownKey() {
+			return 'wikiaInYourLangNotificationShown' + retrieveLinkTitle() + cacheVersion;
+		}
+
+		function getWIYLMessageKey() {
+			return targetLanguage + 'WikiaInYourLangMessage' + retrieveLinkTitle() + cacheVersion;
+		}
+
+		function getWIYLLinkTitlesKey() {
+			return targetLanguage + 'WikiaInYourLangLinkTitles' + cacheVersion;
+		}
+
+		function saveLinkTitle(linkAddress) {
+			var linkAddressAry = linkAddress.split('/'),
+				listOfCachedTitles = '',
+				linkTitle = linkAddressAry[linkAddressAry.length - 1].length === 0 ? 'main' : linkAddressAry[linkAddressAry.length - 1];
+
+			listOfCachedTitles = cache.get(getWIYLLinkTitlesKey());
+			if ( listOfCachedTitles ) {
+				listOfCachedTitles[articleTitle] = linkTitle;
+			} else {
+				listOfCachedTitles = {};
+				listOfCachedTitles[articleTitle] = linkTitle;
+			}
+
+			cache.set(getWIYLLinkTitlesKey(),listOfCachedTitles, cache.CACHE_LONG);
+		}
+
+		function retrieveLinkTitle() {
```

Comment:
I should have added comments. 
Article title is a current article title a user is looking at. LinkTitle is wikia-in-your-lang link's article title, which may or may not be the same as the current article title. Cache is used to store the mapping of article title => link title

Example) 
Suppose there are starwars.wikia.com, and ja.starwars.wikia.com, and a user's language is Japanese, and the user visits English starwars wikia. (sounds already complicated!)
1) User is in main page -> we show 'wikia-in-your-lang' with link 'ja.starwars.com', article title is 'main', link title is 'main'.
2) User is in Luke Skywalker article page ->  we show 'wikia-in-your-lang' with link 'ja.starwars.com/wiki/Luke_Skywalker' since there is an corresponding article in Japanese, article title is 'Luke_Skywalker', link title is 'Luke_Skywalker'.
3) User is in Crossroads article page -> We show 'wikia-in-your-lang' with link 'ja.starwars.com' because there is no corresponding article 'Crossroads' in Japanese. Article is 'Crossroadsd', link title is 'main'.

listOfCachedTitles is a mapping of article title => link title. It should look like {'main':'main', 'Luke_skywalker':'Luke_skywalker', 'Crossroad':'main'}.